src/DataFilterer.ts

```diff
@@ -79,12 +90,12 @@ export class DataFilterer {
             totals.push(total);
         }
 
-        return [datasets, compVars, totals];
+        return {datasets, compVars, totals};
```

Comment:
Just an opportunistic refactor - this seems like a case where type definitions can really make our lives easier!